### ISSUE TITLE
Layout/RedundantLineBreak handles blocks with rescues wrong when InspectBlocks true

### ISSUE THREAD

**Issue by oscaralanpierce** [rubocop/rubocop#15001]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Layout/RedundantLineBreak

### Steps to reproduce the problem

1. Create a Ruby file with the following:

```ruby
array = %w(a b c)

array.each do |element|
  puts element
rescue StandardError
  next
end
```

2. Run Layout/RedundantLineBreak cop with `InspectBlocks` set to `true`
3. See the following output:

```ruby
array.each do |element| puts element rescue StandardError next end
```

### Expected behavior

I expected this block to be unchanged since it has a `rescue`:

```ruby
array.each do |element|
  puts element
rescue StandardError
  next
end
```

### Actual behavior

The block was reformatted like this, breaking interpretation:

```ruby
array.each do |element| puts element rescue StandardError next end
```

### RuboCop configuration

```yaml
Layout/RedundantLineBreak:
  Enabled: true
  InspectBlocks: true
```

### RuboCop version

```shell
1.85.1 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.0, analyzing as Ruby 4.0, running on ruby 4.0.1) [arm64-darwin25]
  - rubocop-rails 2.34.3
  - rubocop-rspec 3.9.0
  - rubocop-performance 1.24.0
  - rubocop-factory_bot 2.28.0
```
